### ISSUE TITLE
Oddities with xlink:title warning for svg

### ISSUE THREAD

**Issue by mattgarrish** [IDPF/epubcheck#629]
Another of the accessibility errors was brought to my attention last week at the 3.1 f2f: if an svg includes an <a> element and doesn't have an @xlink:title, you get the ACC_011 warning message.

This error was confusing to the person because the warning says "Link elements ..." and that was being read literally as <link> and not <a> ("hyperlinks" might be clearer).

But aside from that, I've noticed some other irregularities in validation:
- you get the error even if the a element has text content, which is preferable to the attribute
- you get the error even if the a element has a child title, which is also preferable to the attribute
- the error only occurs when the svg is embedded in an xhtml page. if the svg code is in a standalone file, this warning doesn't get reported

I'm not against reporting this warning, per se, but I think the validation/reporting needs some improvement.

**Comment by rdeltour** [comment]
So, to give context to this warning: the [Content Doc spec says](http://www.idpf.org/epub/301/spec/epub-contentdocs.html#confreq-cd-svg-accessibility):

> "An SVG Document should adhere to the accessibility guidelines given in SVG Access."

And then [SVG Access says](http://www.w3.org/TR/SVG-access/#Links):

>  "The a element should include an xlink:title attribute that explains the target of the link and appropriate title and desc elements should describe the graphical object that is being linked to the target."

@mattgarrish can you please clarify?

>  * you get the error even if the a element has text content, which is preferable to the attribute
>  * you get the error even if the a element has a child title, which is also preferable to the attribute

this makes sense, but on the other hand it seems to contradict SVG access which does not mention these as alternative to `xlink:title`.
It seems we shouldn't make a strict interpretation of SVG access here, and remove the warning in the two cases above. Can you confirm?

> * the error only occurs when the svg is embedded in an xhtml page. if the svg code is in a standalone file, this warning doesn't get reported

OK.

**Comment by mattgarrish** [comment]
Well, those were written 15 years ago when using @title on a elements still had currency, too.

The newest version of SVG is moving away from xlink attributes and has this advice for xlink:title:

Deprecated attribute to describe the meaning of a link or resource in a human-readable fashion. New content should use a ‘title’ child element rather than a ‘xlink:title’ attribute.[1]

Even though it's still a draft, I'd be inclined to follow that guidance given that it's compatible with 1.1.

[1] http://www.w3.org/TR/SVG2/linking.html#XLinkRefAttrs

**Comment by rdeltour** [comment]
> Even though it's still a draft, I'd be inclined to follow that guidance given that it's compatible with 1.1.

Yep, I agree. Is `text` child acceptable too, or would it raise a warning? In my understanding, `title` is really needed when the link is around graphic content, but not around text, right?

**Comment by mattgarrish** [comment]
I think I was making too big an assumption. It's image objects that are the most inaccessible, but having a text child alone may not sufficiently describe what that text links to. Probably best to stick to what we can cite.

**Comment by rdeltour** [comment]
> Probably best to stick to what we can cite.

OK. So it will issue a warning if no `xlink:title` or no `title` child.
